src/components/spreadsheet/equipment-table.js

```diff
@@ -64,14 +131,24 @@ export const EquipmentTable = ({
         []
     );
 
+    const openGeneratorPopup = () => {
+        setPopupEditRegulatingTerminal(true);
+    };
     const gridContext = useMemo(() => {
         return {
             network: network,
             editErrors: {},
             dynamicValidation: {},
             isEditing: topPinnedData ? true : false,
+            handleCellClick: {
+                //functions for handling cell click
+                openGeneratorDialog: () => {
+                    openGeneratorPopup();
+                },
+            },
         };
     }, [network, topPinnedData]);
+    const cellClickedInitialData = gridContext.dynamicValidation;
```

Comment:
you probablly dont need to create this variable, you can pass gridContext.dynamicValidation, to  the pop-up